providers/azure.go

```diff
@@ -313,6 +361,77 @@ func makeAzureHeader(accessToken string) http.Header {
 	return makeAuthorizationHeader(tokenTypeBearer, accessToken, nil)
 }
 
+func (p *AzureProvider) getGroupsFromProfileAPI(ctx context.Context, s *sessions.SessionState) ([]string, error) {
+	if s.AccessToken == "" {
+		return nil, fmt.Errorf("missing access token")
+	}
+
+	groupsURL := getMicrosoftGraphGroupsURL(p.GroupField).String()
+
+	// Need and extra header while talking with MS Graph. For more context see
+	// https://docs.microsoft.com/en-us/graph/api/group-list-transitivememberof?view=graph-rest-1.0&tabs=http#request-headers
+	extraHeader := makeAzureHeader(s.AccessToken)
+	extraHeader.Add("ConsistencyLevel", "eventual")
+
+	var groups []string
+
+	for groupsURL != "" {
+		// logger.Printf("Calling Group API: %s", groupURL)
+		jsonRequest, err := requests.New(groupsURL).
+			WithContext(ctx).
+			WithHeaders(extraHeader).
+			Do().
+			UnmarshalJSON()
+		if err != nil {
+			return nil, err
+		}
+		groupsURL, err = jsonRequest.Get("@odata.nextLink").String()
+		if err != nil {
+			groupsURL = ""
+		}
+		groupsPage, err := getGroupsFromJSON(jsonRequest, p.GroupField)
+		if err != nil {
+			return nil, err
```

Comment:
Wrap this and other errors please

Reply:
Removed it as `getGroupsFromJSON` was returning only nil as error so refactored it.